### PR TITLE
Saleor 1945 adjust adress management to the designs

### PR DESCRIPTION
**Pull request by MarekChoinski**
I want to merge this change because... it:
- fixes styles on address `dashboard/customers/[ID]/addresses`
- moves `Delete` and `Edit` to dropdown menu
- adds `phone` field
- removes some `maybe()`

**PR intended to be tested with API branch:** 

### Screenshots

Before:
image

After:
image

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [ ] Number of API calls is optimized.
1. [ ] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [ ] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1945 Adjust Address management to the designs](https://app.clickup.com/t/2549495/SALEOR-1945)

File: src/customers/components/CustomerAddress/CustomerAddress.tsx [hunk 1]
```diff
@@ -99,6 +96,20 @@ const CustomerAddress: React.FC<CustomerAddressProps> = props => {
                   description: "button"
                 }),
                 onSelect: () => onSetAsDefault(AddressTypeEnum.BILLING)
+              },
+              {
+                label: intl.formatMessage({
```
Comment (orzechdev) [inline src/customers/components/CustomerAddress/CustomerAddress.tsx:101]: We may even move all messages into separate `const messages = defineMessages({...})`

**Review by orzechdev — APPROVED** [review]

File: src/customers/components/CustomerAddressListPage/CustomerAddressListPage.tsx [hunk 2]
```diff
@@ -36,10 +36,9 @@ const useStyles = makeStyles(
       width: 600
     },
     root: {
-      columnGap: theme.spacing(3),
       display: "grid",
-      gridTemplateColumns: "repeat(3, 1fr)",
-      rowGap: theme.spacing(3)
+      gap: `${theme.spacing(3)}px`,
+      gridTemplateColumns: "repeat(3, 1fr)"
```
Comment (orzechdev) [inline src/customers/components/CustomerAddressListPage/CustomerAddressListPage.tsx:41]: Oh, it's not mentioned in task, but I see it looks terrible on mobile screen. May we also add `grid-template-columns: repeat(1, 1fr);` and/or `grid-template-columns: repeat(2, 1fr);` for smaller resolutions?

**Review by orzechdev — APPROVED** [review]
👍

File: src/customers/components/CustomerAddress/CustomerAddress.tsx [hunk 3]
```diff
@@ -25,6 +22,34 @@ export interface CustomerAddressProps {
   onSetAsDefault: (type: AddressTypeEnum) => void;
 }
 
+const messages = defineMessages({
```
Comment (mmarkusik) [inline src/customers/components/CustomerAddress/CustomerAddress.tsx:25]: ✨

**Review by mmarkusik — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**